flox-bash/lib/utils.sh

```diff
@@ -112,17 +112,28 @@ function trace() {
 	# Redirect the output of set -x to /dev/null
 	exec 9>/dev/null
 	local BASH_XTRACEFD=9
-	[ $debug -gt 0 ] || return 0
+	[[ "${debug:-0}" -gt 0 ]] || return 0
 	echo -e "trace:${filecolor}${BASH_SOURCE[2]}:${BASH_LINENO[1]}${colorReset} ${funccolor}${FUNCNAME[1]}${colorReset}( ${argscolor}"$(pprint "$@")"${colorReset} )" 1>&2
 }
 
 # Track exported environment variables for use in verbose output.
+# XXX: This routine is effectively skipped when running `flox' from the
+# nix store, which will instead hard code these values at build time in the
+# generated file `flox-bash/lib/progs.sh'.
+# That generated file sets the variable `_PROGS_INJECTED' which causes this
+# routine to bail early.
+# Any changes to this function should likely be reflected in
+# `pkgs/flox-bash/default.nix'.
 declare -A exported_variables
 function hash_commands() {
 	trace "$@"
+	if [[ -n "${_PROGS_INJECTED:-}" ]]; then
+		return 0;
+	fi
 	set -h # explicitly enable hashing
-	local PATH=@@FLOXPATH@@:$PATH
-	for i in $@; do
+	_OLD_PATH="$PATH"
+	PATH="@@FLOXPATH@@:$PATH"
```

Comment:
So this codepath is skipped if we're not running from the nix store, right? Then how does `@@FLOXPATH@@` get substituted?

If we drop all the `progDecls` stuff in `defualt.nix`, will everything still work correctly because of the change here to `PATH`?

Reply:
reverted.